book-content/chapters/02-ide-superpowers.md

```diff
@@ -0,0 +1,579 @@
+
+## Navigating with Go To Definition and Go To References
+
+The TypeScript server also provides the ability to navigate to the definition of a variable or declaration.
+
+In VS Code, this "Go to Definition" shortcut is used with `Command + Click` on a Mac or `F12` on Windows for the current cursor position. You can also right click and select "Go to Definition" from the context menu on either platform. For the sake of brevity, we'll use the Mac shortcut.
+
+Once you've arrived at the definition location, repeating the `Command + Click` shortcut will show you everywhere that the variable or declaration is used. This is called "Go To References". This is especially useful for navigating around large codebases.
+
+The shortcut can also be used for finding the type definitions for built-in types and libraries. For example, if you `Command + Click` on `document` when using the `getElementById` method, you'll be taken to the type definitions for `document` itself.
+
+This is a great feature for understanding how built-in types and libraries work.
+
+## Rename Symbol
+
+In some situations, you might want to rename a variable across your entire codebase. Let's imagine that a database column changes from 'id' to 'entityId'. A simple find and replace won't work, because 'id' is used in many places for different purposes.
+
+A TypeScript-enabled feature called 'Rename Symbol' allows you to do that with a single action.
+
+Let's take a look at an example.
+
+```typescript
+const filterUsersById = (id: string) => {
+  return users.filter((user) => user.id === id);
+};
+```
+
+Right click on the `id` parameter of the `findUsersById` function and select "Rename Symbol".
+
+A panel will be displayed that prompts for the new name. Type in `userIdToFilterBy` and hit `enter`. VS Code is smart enough to realize that we only want to rename the `id` parameter for the function, and not the `user.id` property:
+
+```typescript
+const filterUsersById = (userIdToFilterBy: string) => {
+  return users.filter((user) => user.id === userIdToFilterBy);
+};
+```
+
+The rename symbol feature is a great tool for refactoring code, and it works across multiple files as well.
+
+## Automatic Imports
+
+Large JavaScript applications can be composed of many, many modules. Manually importing from other files can be tedious. Fortunately, TypeScript supports automatic imports.
+
+When you start typing the name of a variable you want to import, TypeScript will show a list of suggestions when you use the `Ctrl + Space` shortcut. Select a variable from the list, and TypeScript will automatically add an import statement to the top of the file.
+
+You do need to be a little bit careful when using autocompletion in the middle of a name since the rest of the line could be unintentionally altered. To avoid this issue, make sure your cursor is at the end of the name before hitting `Ctrl + Space`.
```

Comment:
This is configurable with `insertMode: insert/replace` - so depending on your configuration it might behave differently if you select a completion when ur cursor is in the middle of a word.

It's not unique to auto-imports though so perhaps I'd move this whole information to a different section